File: ContainsDuplicate.py
# Contains Duplicate
# - Given an integer array `nums`, return True if any value appears at least twice.
# - Return False if every element is distinct.
# - Constraints:
#   - 1 <= nums.length <= 10^5
#   - -10^9 <= nums[i] <= 10^9

from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums) != len(set(nums))

File: ContainsDuplicate.cs
// Contains Duplicate
// - Given an integer array 'nums', return true if any value appears at least twice.
// - Return false if every element is distinct.
// - Constraints:
//   - 1 <= nums.Length <= 100000
//   - -10^9 <= nums[i] <= 10^9

using System;
using System.Collections.Generic;

class Solution
{
    public bool ContainsDuplicate(int[] nums)
    {
        HashSet<int> seen = new HashSet<int>();
        
        foreach (int num in nums)
        {
            if (seen.Contains(num))
            {
                return true; 
            }
            seen.Add(num);
        }

        return false; 
    }
}
